File: Leetcode 3042/3042. Count Prefix and Suffix Pairs I.js
/**
 * @param {string[]} words
 * @return {number}
 */

function isPrefixAndSuffix(s1 , s2){
    if(s1.length > s2.length) return false;
    let n1 = s1.length - 1
    let n2 = s2.length - 1
    for(let i = 0 ; i <= n1 ;i++ ){
        if(s1[i] != s2[i] || s1[n1 - i] != s2[n2 - i]) return false;
    }

    return true;
}
var countPrefixSuffixPairs = function(words) {
    let result = 0;
    let n = words.length
    for(let i = 0; i < n ; i++){
        for(let j = i+1 ; j < n ; j++){
            if(isPrefixAndSuffix(words[i] , words[j])){
                result += 1;
            }
        }
    }
    return result;
    
};

File: Leetcode 3042/3042. Count Prefix and Suffix Pairs I.cpp
class Solution {
public:
    bool isPrefixAndSuffix(string &s1 , string &s2){
        if(s1.length() > s2.length()) return 0;
        int n1 = s1.size()-1 , n2 = s2.size() - 1;
        
        for(int i = 0;i < s1.size() ; i++){
            if(s1[i] != s2[i] || s1[n1 - i] != s2[n2 - i]) return 0;
        }
        return true;
    }
    int countPrefixSuffixPairs(vector<string>& words) {
        int result = 0 , n = words.size();
        for(int i = 0;i < n ; i++){
            for(int j = i+1; j < n ; j++){
                
                result += isPrefixAndSuffix(words[i] , words[j]);
            }
        }
        return result;
    }
};
